src/components/MeasurementOption/MeasurementOption.js

```diff
@@ -35,59 +44,71 @@ class MeasurementOption extends React.PureComponent {
     openMeasurementDropdown: PropTypes.number,
   };
 
-  scaleFromRef = React.createRef();
-
-  scaleToRef = React.createRef();
+  onScaleChange = (value, type) => {
+    this.setState({ [type]: Number(value) }, () => {
+      this.props.onStyleChange('Scale', [
+        [this.state.currScaleFrom, this.state.currUnitFrom],
+        [this.state.currScaleTo, this.state.currUnitTo],
+      ]);
+    });
+  }
 
-  /**
-   * A callback function that is used to change the scale for a measurement tool
-   * The argument has the same format as this.props.scale
-   * If values in the array are falsy then current value will be used instead
-   */
-  onScaleChange = ([[scaleFrom, unitFrom], [scaleTo, unitTo]]) => {
-    const { scale } = this.props;
+  onPrecisionChange = (value, type) => {
+    this.setState({ [type]: Number(value) }, () => {
+      this.props.onStyleChange('Precision', this.state.currPrecision);
+    });
+  }
 
-    scaleFrom = scaleFrom || scale[0][0];
-    unitFrom = unitFrom || scale[0][1];
-    scaleTo = scaleTo || scale[1][0];
-    unitTo = unitTo || scale[1][1];
+  onUnitChange = (event, type) => {
+    this.setState({ [type]: event.target.value }, () => {
+      this.props.onStyleChange('Scale', [
+        [this.state.currScaleFrom, this.state.currUnitFrom],
+        [this.state.currScaleTo, this.state.currUnitTo],
+      ]);
+    });
+  }
 
-    this.props.onStyleChange('Scale', [
-      [scaleFrom, unitFrom],
-      [scaleTo, unitTo],
-    ]);
-    this.props.onOpenDropdownChange(-1);
-  };
+  getLanguage = () => {
+    let lang = 'en';
 
-  onPrecisionChange = precision => {
-    this.props.onStyleChange('Precision', precision);
-    this.props.onOpenDropdownChange(-1);
-  };
+    if (i18next.language) {
+      lang = i18next.language;
+    }
 
-  onBlur = () => {
-    const scaleFromRefValue = this.scaleFromRef.current.value;
-    const scaleToRefValue = this.scaleToRef.current.value;
-    const [[scaleFrom], [scaleTo]] = this.props.scale;
+    return lang;
+  }
 
-    if (scaleFromRefValue === '') {
-      this.scaleFromRef.current.value = scaleFrom;
-    } else if (scaleToRefValue === '') {
-      this.scaleToRef.current.value = scaleTo;
+  formatValue = value => {
```

Comment:
Can we use the same coding style as `getLanguage` for this function? Something like:
```javascript
  formatValue = value => {
    const lang = this.getLanguage();

    if (lang === 'de') {
      value = value.toLocaleString('de-DE', { maximumFractionDigits: 4 });
    } else if (lang === 'fr') {
      value = value.toLocaleString('fr-FR', { maximumFractionDigits: 4 });
    } else if (lang === 'ru') {
      value = value.toLocaleString('ru-RU', { maximumFractionDigits: 4 });
    }

    return value;
  }
```

Reply:
Resolved in 084caf3. Thanks!